pkg/controller/podautoscaler/horizontal_test.go

```diff
@@ -360,4 +366,64 @@ func TestEventNotCreated(t *testing.T) {
 	tc.runTest(t)
 }
 
-// TODO: add more tests
+// TestComputedToleranceAlgImplementation is a regression test which
+// back-calculates a minimal percentage for downscaling based on a small percentage
+// increase in pod utilization which is calibrated against the tolerance value.
+func TestComputedToleranceAlgImplementation(t *testing.T) {
```

Comment:
Please add one more case: if the change is slightly lower than the tolerance, hpa does not rescale. This is needed to prevent failures like https://github.com/kubernetes/kubernetes/pull/18315#issuecomment-164263075.